src/openttd.cpp

```diff
@@ -1441,23 +1445,31 @@ void StateGameLoop()
 	assert(IsLocalCompany());
 }
 
-static IntervalTimer<TimerGameCalendar> _autosave_interval({TimerGameCalendar::MONTH, TimerGameCalendar::Priority::AUTOSAVE}, [](auto)
+/** Interval for regular autosaves. Initialized at zero to disable till settings are loaded. */
+static IntervalTimer<TimerGameAutosave> _autosave_interval(std::chrono::milliseconds::zero(), [](auto)
 {
-	if (_settings_client.gui.autosave == 0) return;
-	if ((_cur_month % _autosave_months[_settings_client.gui.autosave]) != 0) return;
-
 	_do_autosave = true;
 	SetWindowDirty(WC_STATUS_BAR, 0);
+
+	static FiosNumberedSaveName _autosave_ctr("autosave");
```

Comment:
I have no clue, but I think two things worth mentioning in this context:

- The timer runs in the game-thread; as such, this can only be accessed by that one.
- The autosave starts its own thread and copies this information in there. So when the function returns, this variable is no longer used.

So I think we are good here :)